test/hts-precompile/token-managment/tokenManagmentContract.js

```diff
@@ -254,4 +255,693 @@ describe("TokenManagmentContract tests", function () {
         const receiptAssociate = await txAssociate.wait();
         expect(receiptAssociate.events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
     });
+
+    describe('Extended update token info and keys test suite', function() {
+        async function getTokenInfo(contract, token) {
+            const txBeforeInfo = await contract.getTokenInfoPublic(token);
+            const tokenInfo = ((await txBeforeInfo.wait()).events.filter(e => e.event === 'TokenInfo')[0].args.tokenInfo)[0];
+            expect((await txBeforeInfo.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.eq(TX_SUCCESS_CODE);
+            return tokenInfo
+        };
+
+        async function updateTokenInfo(contract, token, updateInfo) {
+            const txUpdate = await contract.updateTokenInfoPublic(token, updateInfo);
+            expect((await txUpdate.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.be.equal(TX_SUCCESS_CODE);
+        }
+
+        function updateTokenInfoValues(keyValueType, key) {
+            const updatedKey = [
+                false,
+                '0x0000000000000000000000000000000000000000',
+                '0x',
+                '0x',
+                '0x0000000000000000000000000000000000000000',
+            ];
+
+            switch (keyValueType) {
+                case utils.KeyValueType.CONTRACT_ID:
+                    updatedKey[1] = key;
+                    break;
+                case utils.KeyValueType.SECP256K1:
+                    updatedKey[3] = key;
+                    break;
+                case utils.KeyValueType.DELEGETABLE_CONTRACT_ID:
+                    updatedKey[4] = key;
+                    break;
+                default:
+                    break;
+            }
+
+            return updatedKey;
+        }
+
+        describe('Admin key set to ECDSA_secp256k', function() {
+            before(async function () {
+                tokenAddress = await utils.createFungibleTokenWithSECP256K1AdminKeyAssociateAndTransferToAddress(tokenCreateContract, tokenCreateContract.address, utils.getSignerCompressedPublicKey());
+                tokenInfoBefore = await getTokenInfo(tokenQueryContract, tokenAddress);
+        
+                await utils.associateToken(tokenCreateContract, tokenAddress, 'TokenCreateContract');
```

Comment:
nit: is this associate and grantKyc necessary as it is being done in the contract?

Reply:
This is a util function, which grants KYC and associate signers[0] and signer[1] to the token, which is needed for the tests.